src/libmongoc/tests/unified/operation.c

```diff
@@ -2761,6 +2764,46 @@ operation_assert_number_connections_checked_out (test_t *test,
    return true;
 }
 
+static bool
+operation_rename (test_t *test,
+                  operation_t *op,
+                  result_t *result,
+                  bson_error_t *error)
+{
+   // First validate the arguments
+   const char *object = op->object;
+   bson_parser_t *bp = bson_parser_new ();
+   char *new_name;
+   bson_parser_utf8 (bp, "to", &new_name);
+   bool parse_ok = bson_parser_parse (bp, op->arguments, error);
+   bson_parser_destroy (bp);
+   if (!parse_ok) {
+      return false;
+   }
+
+   // Now get the entity
+   entity_t *ent = entity_map_get (test->entity_map, object, error);
+   if (!ent) {
+      return false;
+   }
+   // We only support collections so far
+   if (0 != strcmp (ent->type, "collection")) {
+      test_set_error (
+         error,
+         "'rename' is only supported for collection objects '%s' has type '%s'",
+         object,
+         ent->type);
+      return false;
+   }
+   // Rename the collection in the server,
+   mongoc_collection_t *coll = ent->value;
+   if (!mongoc_collection_rename (coll, coll->db, new_name, false, error)) {
```

Comment:
```suggestion
   if (!mongoc_collection_rename (coll, NULL, new_name, false, error)) {
```

Passing `coll->db` reports a use-after-free from ASAN. `coll->db` is freed, than overwritten by the `new_db` [argument here](https://github.com/mongodb/mongo-c-driver/blob/1a97d00d91e7251482ffa76c9b4e854a384d9ce9/src/libmongoc/src/mongoc/mongoc-collection.c#L3192).

Since `coll->db` is not publicly accessible, this may not be a concern for users. Though a possibly more robust solution is to check whether `coll->db == new_db` in `mongoc_collection_rename_with_opts`:
```c
      if (new_db && new_db != collection->db) {
         bson_free (collection->db);
         collection->db = bson_strdup (new_db);
      }
```